hack/runtime-migrator/internal/comparator/comparator.go

```diff
@@ -0,0 +1,54 @@
+package comparator
+
+import (
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/kyma-project/infrastructure-manager/hack/shoot-comparator/pkg/shoot"
+)
+
+type Result struct {
+	Equal bool
+	Diff  []Difference
+}
+
+type Difference struct {
+	ShootName  string
+	LeftShoot  v1beta1.Shoot
+	RightShoot v1beta1.Shoot
+	Message    string
+}
+
+func CompareShoots(leftShoot, rightShoot v1beta1.Shoot) (Result, error) {
+
+	differences, err := compare(leftShoot, rightShoot)
+	if err != nil {
+		return Result{}, err
+	}
+
+	equal := len(differences) == 0
+
+	return Result{
+		Equal: equal,
+		Diff:  differences,
+	}, nil
```

Comment:
`equal` could be a method that reutns `len(this.differences) == 0`, but this is fine too, although easier to break